packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -426,24 +432,93 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
         }
     }
 
+    private tooltip(x: number, y: number): void {
+        const xPos = this.positionXMove;
+        const timeForX = this.getTimeForX(xPos);
+        let timeLabel: string | undefined = timeForX.toString();
+        if (this.props.unitController.numberTranslator) {
+            timeLabel = this.props.unitController.numberTranslator(timeForX);
+        }
+        const chartWidth = this.lineChartRef.current.chartInstance.width;
+        const arraySize = this.state.xyData.labels.length;
+        const index = Math.max(Math.round((xPos / chartWidth) * (arraySize - 1)), 0);
+        const points: any = [];
+        let zeros = 0;
+        this.state.xyData.datasets.forEach((d: any) => {
+            // In case there are less data points than pixels in the chart,
+            // calculate nearest value.
+            const adjustedY = d.data[index];
+            const rounded = isNaN(adjustedY) ? 0 : (BIMath.round(Number(adjustedY) * 100)/BIMath.round(100));
```

Comment:
BIMath returns bigint integers, so it loses all fractional part after the division. I think the intention was to keep two decimal places? You could use Number.toFixed(2), perhaps conditionally if the number has a fractional part (can be checked with % 1 != 0 perhaps ?).

Reply:
True. I used BIMath when TypeScript complained about assigning values at some point, but that requirement is no longer there when I added a new interface in tooltip-xy-component.
But I prefer to use `(Number(adjustedY) * 100) / 100` instead of `toFixed(2)` just because the former strips decimal numbers when they are 0 (it shows `100` , not `100.00`).